### PR TITLE
wip: dataloader first draft

### PR DESCRIPTION
**Pull request by ssenan**

### CODE REVIEW

**Comment by mateibejan1** [comment]
Could you just change the name of the file to sequence_dataloader.py, so we can differentiate from future possible loaders?

**Comment by lucapinello** [comment]
Thanks guys!

Two comments:

1. I am fine to have separate files, @mateibejan1 however we can also have different classes on this module and keep all the related classes and functions in a small number of files. I am envisioning to do the same for the metrics e.g. we can have a single file called metrics and we can import from there. 
2. @ssenan I think it may be better to not use the one-hot encoding for the components and using a simple integer so we can keep memory under control when we will load the entire 3.5M sequences!

**Comment by mateibejan1** [comment]
I understand you perspective @lucapinello. I would only like to apply this separation-by-class to diffusion code, denoising networks and dataloaders. The reasoning is that the code for these types of files tends to get repetitive and once you have 3-4 modules in the same file it becomes tiring to keep track of which section you're debugged or changed. I am ok with keeping all metrics, utils etc. in a small number of files in their designated directories.

File: models/vanilla_diffusion/dataloader.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        super().__init__()
+        self.train_path = train_path
+        self.val_path = val_path
+        self.test_path = test_path
+        self.transform = transform
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+
+
+    def prepare_data(self):
```
Comment (SauravMaheshkar) [inline models/vanilla_diffusion/dataloader.py:61]: We should probably drop this redundant function definition.

File: models/vanilla_diffusion/dataloader.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import pandas as pd
+import numpy as np
+
+import torch
+import torchvision.transforms as T
+import torch.nn.functional as F
+import pytorch_lightning as pl 
+from torch.utils.data import Dataset, DataLoader 
+
+class SequenceDataset(Dataset):
```
Comment (SauravMaheshkar) [inline models/vanilla_diffusion/dataloader.py:10]: Proposition: We could also separate this script into a `dataloader.py` and a `dataset.py`.

The `dataset.py` would contain the `SequenceDataset` Class and the `dataloader.py` can have the `SequenceDataModule` Class (or a `BaseDataModule` and further subclasses such as `SequenceDataModule` as suggested by @lucapinello)

**Review by SauravMaheshkar — CHANGES_REQUESTED** [review]

**Comment by ssenan** [comment]
@mateibejan1 Thanks for the suggestion. Can you help me better understand an example file naming breakdown you had in mind in regards to the entire project?

My thought process was more inline with Luca's where all loaders can be kept in a single dataloader.py file, and they can then be called in a main script using something like "from dataloader import SequenceDataModule". This is in line with some other implementations using medical image data that I have seen.

I'm also available on  discord @ssenan

**Comment by IhabBendidi** [comment]
There is currently a discussion on folder structure of the code base in #19 , referencing it here so that you guys can check it and add your ideas

**Pull request merged**